src/safeposix/syscalls/fs_calls.rs

```diff
@@ -2608,8 +2608,8 @@ impl Cage {
                             }
                             if dir_inode_obj.linkcount == 2 && dir_inode_obj.refcount == 0 {
                                 //removing the file from the metadata
-                                FS_METADATA.inodetable.remove(&inodenum);
                                 drop(inodeobj);
+                                FS_METADATA.inodetable.remove(&inodenum);
```

Comment:
I know this isn't your syscall to do but a comment here would be helpful for the next person.